lib/common/Makefile.am

```diff
@@ -92,6 +94,13 @@ libcrmcommon_la_SOURCES	+= xpath.c
 # file, which may have already been cleaned.
 nodist_libcrmcommon_la_SOURCES	= md5.c
 
+libcrmcommon_test_la_SOURCES	= $(libcrmcommon_la_SOURCES)
+libcrmcommon_test_la_SOURCES	+= mock.c
+libcrmcommon_test_la_LDFLAGS	= $(LDFLAGS_HARDENED_LIB) $(WRAPPED_FLAGS)
+libcrmcommon_test_la_CFLAGS	= $(libcrmcommon_la_CFLAGS)
+libcrmcommon_test_la_LIBADD	= $(libcrmcommon_la_LIBADD)
+nodist_libcrmcommon_test_la_SOURCES = $(nodist_libcrmcommon_la_SOURCES)
```

Comment:
Can we move this to the tests subdirectory and make it a dependency of make check, so it only gets built if someone runs that target? Most users building from source aren't using the pacemaker rpm target

Reply:
Yes maybe, if I can figure out how to set the variables to be the same as the libcrmcommon_* ones up a directory.  There's also a check_LTLIBRARIES variable that's only supposed to build libraries when you run "make check", but I haven't been able to make it work yet.

Reply:
Okay, there we go.  libcrmcommon_test.la is now only built when "make check" is run.  Is that good enough, or should I try to make this happen in the tests directory as well?

Reply:
Moving to tests would be nice but not worth a lot of effort